zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,2265 @@
+    evm_circuit::{
+        util::{and, constraint_builder::BaseConstraintBuilder, not, or},
+        witness::{RlpTxTag, RlpWitnessGen, N_TX_TAGS},
+    },
+    table::RlpTable,
+    util::Expr,
+    witness::SignedTransaction,
+};
+
+#[derive(Clone, Debug)]
+/// Config for the RLP circuit.
+pub struct RlpCircuit<F> {
+    /// Denotes whether or not the row is enabled.
+    q_usable: Selector,
+    /// Denotes whether the row is the first row in the layout.
+    is_first: Column<Advice>,
+    /// Denotes whether the row is the last byte in the RLP-encoded data.
+    is_last: Column<Advice>,
+    /// Embedded lookup table in RLP circuit.
+    rlp_table: RlpTable,
+    /// Denotes the index of this row, starting from `1` and ending at `n`
+    /// where `n` is the byte length of the RLP-encoded data.
+    index: Column<Advice>,
+    /// Denotes the index of this row, but reversed. It starts from `n` where
+    /// `n` is the byte length of the RLP-encoded data and ends at `1`.
+    rindex: Column<Advice>,
+    /// Denotes the byte value at this row index from the RLP-encoded data.
+    value: Column<Advice>,
+    /// List of columns that are assigned:
+    /// val := (tag - RlpTxTag::{Variant}).inv()
+    tx_tags: [Column<Advice>; N_TX_TAGS],
+    /// Denotes the current tag's span in bytes.
+    tag_length: Column<Advice>,
+    /// Denotes an accumulator for the length of data, in the case where len >
+    /// 55 and the length is represented in its big-endian form.
+    length_acc: Column<Advice>,
+    /// Denotes an accumulator for the value field over all rows (bytes).
+    value_rlc: Column<Advice>,
+    /// Comparison chip to check: 1 <= tag_index.
+    tag_index_cmp_1: ComparatorConfig<F, 1>,
+    /// Comparison chip to check: tag_index <= tag_length.
+    tag_index_length_cmp: ComparatorConfig<F, 1>,
+    /// Comparison chip to check: 1 <= tag_length.
+    tag_length_cmp_1: ComparatorConfig<F, 1>,
+    /// Lt chip to check: tag_index < 10.
+    tag_index_lt_10: LtConfig<F, 1>,
+    /// Lt chip to check: tag_index < 34.
+    tag_index_lt_34: LtConfig<F, 1>,
+    /// Lt chip to check: 127 < value.
+    value_gt_127: LtConfig<F, 1>,
+    /// Lt chip to check: 183 < value.
+    value_gt_183: LtConfig<F, 1>,
+    /// Lt chip to check: 191 < value.
+    value_gt_191: LtConfig<F, 1>,
+    /// Lt chip to check: 247 < value.
+    value_gt_247: LtConfig<F, 1>,
+    /// Lt chip to check: value < 128.
+    value_lt_129: LtConfig<F, 1>,
+    /// Lt chip to check: value < 184.
+    value_lt_184: LtConfig<F, 1>,
+    /// Lt chip to check: value < 192.
+    value_lt_192: LtConfig<F, 1>,
+    /// Lt chip to check: value < 248.
+    value_lt_248: LtConfig<F, 1>,
+    /// Lt chip to check: value < 256.
+    value_lt_256: LtConfig<F, 2>,
+    /// Comparison chip to check: 0 <= length_acc.
+    length_acc_cmp_0: ComparatorConfig<F, 1>,
+}
+
+impl<F: Field> RlpCircuit<F> {
+    pub(crate) fn configure(meta: &mut ConstraintSystem<F>, r: Expression<F>) -> Self {
```

Comment:
on the naming: https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/650

Reply:
Fixed here: https://github.com/scroll-tech/zkevm-circuits/pull/220/commits/80dae7e7b9592c0b52d641ab000e5bb42958c728